interfaces/builtin/desktop.go

```diff
@@ -370,6 +370,55 @@ dbus (receive)
     member="{AddNotification,RemoveNotification}"
     peer=(label=unconfined),
 
+# Allow registering session with GDM, necessary for screen locking
+dbus (send)
+    bus=system
+    path=/org/gnome/DisplayManager/Manager
+    interface=org.gnome.DisplayManager.Manager
+    member={RegisterSession,OpenReauthenticationChannel}
+    peer=(label=unconfined),
+dbus (send)
+    bus=system
+    path=/org/gnome/DisplayManager/Manager
+    interface=org.freedesktop.DBus.Properties
+    member="Get{,All}"
+    peer=(label=unconfined),
+
+# Allow gnome-shell to bind to its various D-Bus names
+dbus (bind)
+    bus=session
+    name=org.gnome.Mutter.*,
+dbus (bind)
+    bus=session
+    name=org.gnome.Shell{,.*},
+
+# Allow gnome-settings-daemon to bind its various D-Bus names
+dbus (bind)
+    bus=session
+    name=org.gnome.SettingsDaemon{,.*},
+dbus (bind)
+    bus=session
+    name=org.gtk.Settings,
+
+# Allow the shell to communicate with colord
+dbus (send, receive)
+    bus=system
+    path=/org/freedesktop/ColorManager{,/**}
+    interface=org.freedesktop.ColorManager*
+    peer=(label=unconfined),
+dbus (send, receive)
+    bus=system
+    path=/org/freedesktop/ColorManager{,/**}
+    interface=org.freedesktop.DBus.Properties
+    member={Get,GetAll,PropertiesChanged}
+    peer=(label=unconfined),
+
+# Allow access to GDM's private reauthentication channel socket
+unix (connect, receive, send)
+    type=stream
+    peer=(addr="@/tmp/dbus-*"),
```

Comment:
I included rules here to handle the old socket location used in Ubuntu 22.04's GDM (abstract socket named like `/tmp/dbus-*`), and the new socket location in Ubuntu 23.10 plus the MR (non-abstract socket socket named like `/run/gdm3/dbus/dbus-*`). The two rules look different because AppArmor handles abstract and non-abstract UNIX domain sockets differently.

At this point we're unlikely to go back to the old GNOME stack, so could probably just the rule for the abstract socket.

Reply:
Thank you for the explanation, makes total sense now